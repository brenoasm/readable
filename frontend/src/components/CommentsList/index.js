import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as _ from 'lodash';

import CommentRow from '../CommentRow';

const propTypes = {
  comments: PropTypes.arrayOf(PropTypes.object),
  editClick: PropTypes.func,
  deleteComment: PropTypes.func,
  modifyCommentVoteValues: PropTypes.func
};

const defaultProps = {
  comments: [],
  editClick: () => {},
  deleteComment: () => {},
  modifyCommentVoteValues: () => {}
};

const StyledCommentsList = styled.div`
  ul {
    list-style-type: none;
    padding-left: 0;
  }
`;

const CommentsList = ({
  comments,
  editClick,
  deleteComment,
  modifyCommentVoteValues
}) => {
  const sortedComments =  _.sortBy(comments, comment => comment.voteScore).reverse();

  return (
    <StyledCommentsList>
      {Array.isArray(sortedComments) && (
        <ul>
          {sortedComments.map(comment => (
            <li key={comment.id}>
              <CommentRow
                editClick={editClick}
                deleteComment={deleteComment}
                comment={comment}
                modifyCommentVoteValues={modifyCommentVoteValues} />
            </li>
          ))}
        </ul>
      )}
    </StyledCommentsList>
  )
};

CommentsList.propTypes = propTypes;
CommentsList.defaultProps = defaultProps;

export default CommentsList;
