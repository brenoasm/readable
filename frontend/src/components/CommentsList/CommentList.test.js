/* eslint-disable */

import React from 'react';

import CommentsList from './';

import CommentMock from '../../testHelpers/mocks/comments';

describe('CommentsList', () => {
  let wrapper;

  beforeEach(() => {
    const initialState = {
      comments: [
        CommentMock.comments[0],
        CommentMock.comments[1]
      ],
      deleteComment: jest.fn(),
      editClick: jest.fn(),
      modifyCommentVoteValues: jest.fn()
    };

    wrapper = shallow(<CommentsList {...initialState} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render two CommentRow component', () => {
    expect(wrapper.find('CommentRow')).toHaveLength(2);
  })

  it('should order the comments by voteScore desc', () => {
    const commentRows = wrapper.find('CommentRow');

    expect(commentRows.get(0).props.comment.voteScore)
      .toBeGreaterThan(commentRows.get(1).props.comment.voteScore);
  });
});
