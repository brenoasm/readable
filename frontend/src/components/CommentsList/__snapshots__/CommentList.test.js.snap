// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsList matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsList
    comments={
        Array [
            Object {
              "author": "Breno",
              "body": "Test",
              "deleted": false,
              "id": "1",
              "parentDeleted": false,
              "parentId": "1",
              "timestamp": 1538417732043,
              "voteScore": 1,
            },
            Object {
              "author": "Joãozinho",
              "body": "Test 2",
              "deleted": false,
              "id": "2",
              "parentDeleted": false,
              "parentId": "1",
              "timestamp": 1538419906880,
              "voteScore": 2,
            },
          ]
    }
    deleteComment={[Function]}
    editClick={[Function]}
    modifyCommentVoteValues={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ul>
        <li>
                <CommentRow
                        comment={
                                Object {
                                        "author": "Joãozinho",
                                        "body": "Test 2",
                                        "deleted": false,
                                        "id": "2",
                                        "parentDeleted": false,
                                        "parentId": "1",
                                        "timestamp": 1538419906880,
                                        "voteScore": 2,
                                      }
                        }
                        deleteComment={[Function]}
                        editClick={[Function]}
                        modifyCommentVoteValues={[Function]}
                />
        </li>
        <li>
                <CommentRow
                        comment={
                                Object {
                                        "author": "Breno",
                                        "body": "Test",
                                        "deleted": false,
                                        "id": "1",
                                        "parentDeleted": false,
                                        "parentId": "1",
                                        "timestamp": 1538417732043,
                                        "voteScore": 1,
                                      }
                        }
                        deleteComment={[Function]}
                        editClick={[Function]}
                        modifyCommentVoteValues={[Function]}
                />
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <CommentRow
                        comment={
                                    Object {
                                                "author": "Joãozinho",
                                                "body": "Test 2",
                                                "deleted": false,
                                                "id": "2",
                                                "parentDeleted": false,
                                                "parentId": "1",
                                                "timestamp": 1538419906880,
                                                "voteScore": 2,
                                              }
                        }
                        deleteComment={[Function]}
                        editClick={[Function]}
                        modifyCommentVoteValues={[Function]}
            />
</li>,
          <li>
            <CommentRow
                        comment={
                                    Object {
                                                "author": "Breno",
                                                "body": "Test",
                                                "deleted": false,
                                                "id": "1",
                                                "parentDeleted": false,
                                                "parentId": "1",
                                                "timestamp": 1538417732043,
                                                "voteScore": 1,
                                              }
                        }
                        deleteComment={[Function]}
                        editClick={[Function]}
                        modifyCommentVoteValues={[Function]}
            />
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <CommentRow
              comment={
                            Object {
                                          "author": "Joãozinho",
                                          "body": "Test 2",
                                          "deleted": false,
                                          "id": "2",
                                          "parentDeleted": false,
                                          "parentId": "1",
                                          "timestamp": 1538419906880,
                                          "voteScore": 2,
                                        }
              }
              deleteComment={[Function]}
              editClick={[Function]}
              modifyCommentVoteValues={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "comment": Object {
                "author": "Joãozinho",
                "body": "Test 2",
                "deleted": false,
                "id": "2",
                "parentDeleted": false,
                "parentId": "1",
                "timestamp": 1538419906880,
                "voteScore": 2,
              },
              "deleteComment": [Function],
              "editClick": [Function],
              "modifyCommentVoteValues": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <CommentRow
              comment={
                            Object {
                                          "author": "Breno",
                                          "body": "Test",
                                          "deleted": false,
                                          "id": "1",
                                          "parentDeleted": false,
                                          "parentId": "1",
                                          "timestamp": 1538417732043,
                                          "voteScore": 1,
                                        }
              }
              deleteComment={[Function]}
              editClick={[Function]}
              modifyCommentVoteValues={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "comment": Object {
                "author": "Breno",
                "body": "Test",
                "deleted": false,
                "id": "1",
                "parentDeleted": false,
                "parentId": "1",
                "timestamp": 1538417732043,
                "voteScore": 1,
              },
              "deleteComment": [Function],
              "editClick": [Function],
              "modifyCommentVoteValues": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ul>
          <li>
                    <CommentRow
                              comment={
                                        Object {
                                                  "author": "Joãozinho",
                                                  "body": "Test 2",
                                                  "deleted": false,
                                                  "id": "2",
                                                  "parentDeleted": false,
                                                  "parentId": "1",
                                                  "timestamp": 1538419906880,
                                                  "voteScore": 2,
                                                }
                              }
                              deleteComment={[Function]}
                              editClick={[Function]}
                              modifyCommentVoteValues={[Function]}
                    />
          </li>
          <li>
                    <CommentRow
                              comment={
                                        Object {
                                                  "author": "Breno",
                                                  "body": "Test",
                                                  "deleted": false,
                                                  "id": "1",
                                                  "parentDeleted": false,
                                                  "parentId": "1",
                                                  "timestamp": 1538417732043,
                                                  "voteScore": 1,
                                                }
                              }
                              deleteComment={[Function]}
                              editClick={[Function]}
                              modifyCommentVoteValues={[Function]}
                    />
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <CommentRow
                            comment={
                                          Object {
                                                        "author": "Joãozinho",
                                                        "body": "Test 2",
                                                        "deleted": false,
                                                        "id": "2",
                                                        "parentDeleted": false,
                                                        "parentId": "1",
                                                        "timestamp": 1538419906880,
                                                        "voteScore": 2,
                                                      }
                            }
                            deleteComment={[Function]}
                            editClick={[Function]}
                            modifyCommentVoteValues={[Function]}
              />
</li>,
            <li>
              <CommentRow
                            comment={
                                          Object {
                                                        "author": "Breno",
                                                        "body": "Test",
                                                        "deleted": false,
                                                        "id": "1",
                                                        "parentDeleted": false,
                                                        "parentId": "1",
                                                        "timestamp": 1538417732043,
                                                        "voteScore": 1,
                                                      }
                            }
                            deleteComment={[Function]}
                            editClick={[Function]}
                            modifyCommentVoteValues={[Function]}
              />
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <CommentRow
                comment={
                                Object {
                                                "author": "Joãozinho",
                                                "body": "Test 2",
                                                "deleted": false,
                                                "id": "2",
                                                "parentDeleted": false,
                                                "parentId": "1",
                                                "timestamp": 1538419906880,
                                                "voteScore": 2,
                                              }
                }
                deleteComment={[Function]}
                editClick={[Function]}
                modifyCommentVoteValues={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "comment": Object {
                  "author": "Joãozinho",
                  "body": "Test 2",
                  "deleted": false,
                  "id": "2",
                  "parentDeleted": false,
                  "parentId": "1",
                  "timestamp": 1538419906880,
                  "voteScore": 2,
                },
                "deleteComment": [Function],
                "editClick": [Function],
                "modifyCommentVoteValues": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <CommentRow
                comment={
                                Object {
                                                "author": "Breno",
                                                "body": "Test",
                                                "deleted": false,
                                                "id": "1",
                                                "parentDeleted": false,
                                                "parentId": "1",
                                                "timestamp": 1538417732043,
                                                "voteScore": 1,
                                              }
                }
                deleteComment={[Function]}
                editClick={[Function]}
                modifyCommentVoteValues={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "comment": Object {
                  "author": "Breno",
                  "body": "Test",
                  "deleted": false,
                  "id": "1",
                  "parentDeleted": false,
                  "parentId": "1",
                  "timestamp": 1538417732043,
                  "voteScore": 1,
                },
                "deleteComment": [Function],
                "editClick": [Function],
                "modifyCommentVoteValues": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
